### PR TITLE
feat: Mark PartialChunks as non-Cold column and clean-up it from Cold DB

### PR DESCRIPTION
**Pull request by tayfunelmas**
`PartialChunk` is not needed for replaying the chain history and answering historical queries. It can be re-constructed from other information such as `Chunks` column. Thus, we remove it from Cold DB. This will save ~9 TB disk space for archival nodes ([dashboard](https://grafana.nearone.org/d/bdm279s7fsqgwc/disk-usage-stats?orgId=1&from=now-90d&to=now&var-chain_id=mainnet&var-node_id=rc-mainnet-canary-rpc-01-us-central1-a-2aaa0c29&var-disk_device=sdb&var-disk_mount_point=%2Fhome%2Fubuntu%2F.near%2Fdata&viewPanel=panel-18)).

For this, we pass the `DbKind` and whether the node is archival to the migration code and skip the migration for non-Cold DBs. 

Testing: Tested with mainnet archival node, where the node completed the migration in ~600ms and continued operating normally after that.

Saving: 9TB of 60TB total (so 15% of the storage size).

### CODE REVIEW

File: core/store/src/columns.rs [hunk 1]
```diff
@@ -467,6 +465,8 @@ impl DBCol {
             DBCol::LatestWitnessesByIndex => false,
             // Deprecated.
             DBCol::_ReceiptIdToShardId => false,
+            // This can be re-constructed from the Chunks column, so no need to store in Cold DB.
+            DBCol::PartialChunks => false,
```
Comment (Trisfald) [inline core/store/src/columns.rs:469]: Does it work magically work?

I think cold DB will throw either an error or a panic if a non cold column is read. Does it mean this column was never read from cold?

**Comment by Trisfald** [comment]
I'm wondering if we have an archival node with this change applied, that can be used to run some manual checks

File: core/store/src/columns.rs (hunk 1 above)
Comment (tayfunelmas) [reply core/store/src/columns.rs:469]: Yes, it will not be read from cold db, otherwise yes, the binary should panic.

**Comment by tayfunelmas** [comment]
> I'm wondering if we have an archival node with this change applied, that can be used to run some manual checks

I brought up a mainnet archival node and applied the migration, it took ~600ms and the node was able to run after the migration.

**Comment by tayfunelmas** [comment]
> How confident are you that this column is not read from RPC? If not fully you can also try to send some RPC queries to your canary node or even mirror some rpc traffic there.

I traced the code paths reading from this column and could not find one that is used in RPCs. This column is being used in operating the chain from the head.

There is one use for applying chunks in a state-viewer command but the info (receipts) it provides can be taken from the `Chunks` column (previous outgoing receipts).

**Review by Trisfald — APPROVED** [review]
👍

File: nearcore/src/migrations.rs [hunk 2]
```diff
@@ -86,6 +91,11 @@ impl<'a> near_store::StoreMigrator for Migrator<'a> {
             37 => near_store::migrations::migrate_37_to_38(store),
             38 => near_store::migrations::migrate_38_to_39(store),
             39 => near_store::migrations::migrate_39_to_40(store),
+            40 => near_store::migrations::migrate_40_to_41(
+                store,
+                kind.unwrap_or_else(|| panic!("DB kind is not set for DB at version: {}", version)),
```
Comment (wacban) [inline nearcore/src/migrations.rs:96]: nit: `.expect(format!(...))` would be a bit more typical

**Review by wacban — APPROVED** [review]
LGTM

How much savings is that going to give us (absolute and relative)?

File: nearcore/src/migrations.rs (hunk 2 above)
Comment (tayfunelmas) [reply nearcore/src/migrations.rs:96]: I tried using it before, but compiling `expect` in that form fails due to calling a function in expect, so if I need formatting with arguments, I use `unwrap_or_else(panic!(...))`.

**Comment by tayfunelmas** [comment]
> How much savings is that going to give us (absolute and relative)?

It is 9T of 60T total, so 15%. This is the biggest saving other than the State column, and I do not expect such large saving from other columns, unfortunately.

File: nearcore/src/migrations.rs (hunk 2 above)
Comment (wacban) [inline nearcore/src/migrations.rs:96]: I think the following should work:
```suggestion
                kind.expect(&format!("DB kind is not set for DB at version: {}", version)),
```

**Pull request merged**